### PR TITLE
Replaced __destroy__ with proper free() methods.

### PR DESCRIPTION
**Pull request by thomasfanell**
*Replaced __destroy__ with proper free() methods.
*Removed abstract _allocate from base classes.

### CODE REVIEW

**Comment by thomasfanell** [comment]
All whitespace changes should be removed now

File: source/collections/Vector.ooc [hunk 1]
```diff
@@ -37,17 +41,14 @@ Vector: abstract class <T> {
 			}
 		}
 	}
-	_allocate: abstract func (capacity: Int)
-
+
```
Comment (davidhesselbom) [inline source/collections/Vector.ooc]: Hmm, a newline was removed and a newline was added?

File: source/collections/Vector.ooc [hunk 2]
```diff
@@ -79,6 +80,7 @@ Vector: abstract class <T> {
 		else
 			memcpy(destination, source, length)
 	}
+
```
Comment (davidhesselbom) [inline source/collections/Vector.ooc]: This is a new newline

File: source/collections/Vector.ooc [hunk 3]
```diff
@@ -137,13 +142,9 @@ StackVector: class <T> extends Vector<T> {
...
-	// TODO: Why does this function exist here?
-	_allocate: func(capacity: Int)
-
 	resize: override func(capacity: Int) {
 		if (capacity > this capacity)
 			capacity = this capacity
 		super(capacity)
 	}
-}
+}
```
Comment (davidhesselbom) [inline source/collections/Vector.ooc]: Any idea what's going on here?

File: source/collections/Vector.ooc [hunk 4]
```diff
@@ -23,11 +23,15 @@ Vector: abstract class <T> {
...
-		this _allocate(this _capacity)
-		memset(this _backend, 0, capacity * T size)
 		this _freeContent = freeContent
 	}
-	__destroy__: func {	this _free(0, this capacity) }
+	free: override func {
+		this _free(0, this capacity)
+		if (!(this instanceOf?(StackVector))) {
+			gc_free(this _backend)
+		}
```
Comment (davidhesselbom) [inline source/collections/Vector.ooc]: We don't need these brackets. Remove and amend.

File: source/collections/Vector.ooc (hunk 3 above)
Comment (thomasfanell) [inline source/collections/Vector.ooc]: No idea

File: source/collections/Vector.ooc (hunk 2 above)
Comment (davidhesselbom) [inline source/collections/Vector.ooc]: And it shouldn't even be there

**Comment by davidhesselbom** [comment]
Looks good, I think. Just need to test it...

**Comment by davidhesselbom** [comment]
I get
```
(SIGSEGV) segmentation fault
[fancy backtrace]
0     BacktraceHandler backtrace()     in lang/Backtrace    (at /home/dhesselbom/versioned/ooc/ooc-kean/source/sdk/lang/Backtrace.ooc:50)              
1     Exception getCurrentBacktrace()  in lang/Exception    (at /home/dhesselbom/versioned/ooc/ooc-kean/source/sdk/lang/Exception.ooc:208)             
2     _signalHandler()                 in lang/Exception    (at /home/dhesselbom/versioned/ooc/ooc-kean/source/sdk/lang/Exception.ooc:269)             
3     killpg()                         in                   (at (null):0)                                                                              
4     Object free()                    in lang/types        (at /home/dhesselbom/versioned/ooc/ooc-kean/source/sdk/lang/types.ooc:45)                  
5     Vector _free_range()             in Vector            (at /home/dhesselbom/versioned/ooc/ooc-kean/source/collections/Vector.ooc:36)              
6     Vector free_impl()               in Vector            (at /home/dhesselbom/versioned/ooc/ooc-kean/source/collections/Vector.ooc:30)              
7     Vector free()                    in Vector            (at /home/dhesselbom/versioned/ooc/ooc-kean/source/collections/Vector.ooc:30)              
8     VectorList free_impl()           in VectorList        (at /home/dhesselbom/versioned/ooc/ooc-kean/source/collections/VectorList.ooc:75)          
9     VectorList free()                in VectorList        (at /home/dhesselbom/versioned/ooc/ooc-kean/source/collections/VectorList.ooc:35)          
10    GpuImageBin free_impl()          in GpuImageBin       (at /home/dhesselbom/versioned/ooc/ooc-kean/source/draw/gpu/GpuImageBin.ooc:56)            
11    GpuImageBin free()               in GpuImageBin       (at /home/dhesselbom/versioned/ooc/ooc-kean/source/draw/gpu/GpuImageBin.ooc:122)           
12    GpuContext free_impl()           in GpuContext        (at /home/dhesselbom/versioned/ooc/ooc-kean/source/draw/gpu/GpuContext.ooc:36)             
13    OpenGLES3Context free_impl()     in OpenGLES3Context  (at /home/dhesselbom/versioned/ooc/ooc-kean/source/draw/gpu/opengl/OpenGLES3Context.ooc:63)
14    Window free_impl()               in Window            (at /home/dhesselbom/versioned/ooc/ooc-kean/source/draw/gpu/pc/Window.ooc:63)              
15    Window free()                    in Window            (at /home/dhesselbom/versioned/ooc/ooc-kean/source/draw/gpu/pc/Window.ooc:43)              
16    main()                           in                   (at /home/dhesselbom/versioned/ooc/ooc-vidproc/source/vidhance/sdk/pc/main.ooc:128)        
17    libc_start_main()                in                   (at /build/buildd/eglibc-2.19/csu/libc-start.c:321)                                        
18    _start()                         in                   (at (null):0)                                         
```
when running the main project.

**Pull request merged**